authentication/pipeline.py

```
@@ -0,0 +1,65 @@
+            raise ValueError('Invalid audience')
+            
+        if not userinfo.get('email_verified'):
+            raise ValueError('Email not verified with Google')
+            
+        if float(userinfo.get('exp', 0)) < time.time():
+            raise ValueError('Token has expired')
+
+        return userinfo
+        
+    except Exception as e:
+        raise ValueError(f'Token verification failed: {str(e)}')
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)